core/src/main/scala/cromwell/core/PathFactory.scala

```diff
@@ -0,0 +1,110 @@
+package cromwell.core
+
+import java.io.Writer
+import java.nio.file.{FileSystem, Path}
+
+import scala.collection.immutable.Queue
+import scala.util.{Success, Try}
+
+object PathFactory {
+  private def findFileSystem(rawString: String, fss: List[FileSystem], mapping: PartialFunction[FileSystem, Try[Path]]) = {
```

Comment:
If you're going to take in a `List` and immediately transform it to a `Stream`, why take in a `List` in the first place?